docs: add Zeabur as a hosting provider (#1059)
grammyjs/website: site/docs/hosting/zeabur-deno.md

---
prev: false
next: false
---

# Hosting: Zeabur (Deno)

[Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease. It supports various programming languages and frameworks, including Deno and grammY.

This tutorial will guide you how to deploy your grammY bots with Deno to [Zeabur](https://zeabur.com/).

::: tip Looking for the Node.js Version?
This tutorial explains how to deploy a Telegram bot to Zeabur using Deno.
If you're looking for the Node.js version, please check out [this tutorial](./zeabur-nodejs) instead.
:::

## Prerequisites

> To follow along, you need to have a [Github](https://github.com) and [Zeabur](https://zeabur.com/) account.

Reviewer: ```suggestion
To follow along, you need to have [Github](https://github.com) and [Zeabur](https://zeabur.com/) accounts.
```

### Method 1: Create a New Project From Scratch

> Make sure you have Deno installed on your local machine. 

Initialize your project and install some necessary dependencies:

```sh
# Initialize the project.
mkdir grammy-bot
cd grammy-bot

# Create main.ts file
touch main.ts

# Create deno.json file to generate lock file
touch deno.json
```

Then modify `main.ts` file with the following code:

```typescript
import { Bot } from "https://deno.land/x/grammy@v1.23.0/mod.ts";

> const bot = new Bot("your_telegram_bot_token"); 

Reviewer: The Node version reads the token from the environment. Is there a reason why this isn't done here?

---
prev: false
next: false
---

# Hosting: Zeabur (Deno)

> [Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease. It supports various programming languages and frameworks, including Deno and grammY.

Reviewer: Please stick with one sentence per line
```suggestion
[Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease.
It supports various programming languages and frameworks, including Deno and grammY.
```

This tutorial will guide you how to deploy your grammY bots with Deno to [Zeabur](https://zeabur.com/).

::: tip Looking for the Node.js Version?
This tutorial explains how to deploy a Telegram bot to Zeabur using Deno.
If you're looking for the Node.js version, please check out [this tutorial](./zeabur-nodejs) instead.
:::

## Prerequisites

To follow along, you need to have [Github](https://github.com) and [Zeabur](https://zeabur.com/) accounts.

### Method 1: Create a New Project From Scratch

> Make sure you have Deno installed on your local machine.

Initialize your project and install some necessary dependencies:

```sh
# Initialize the project.
mkdir grammy-bot
cd grammy-bot

# Create main.ts file
touch main.ts

# Create deno.json file to generate lock file
touch deno.json
```

Then modify `main.ts` file with the following code:

```typescript

> import { Bot } from "https://deno.land/x/grammy@v1.23.0/mod.ts";

Reviewer: The latest version number will actually be inserted during build. Just use a bare link instead.
```suggestion
import { Bot } from "https://deno.land/x/grammy/mod.ts";
```

const token = Deno.env.get("TELEGRAM_BOT_TOKEN");
if (!token) throw new Error("TELEGRAM_BOT_TOKEN is unset");

const bot = new Bot(token); 

bot.command("start", (ctx) => ctx.reply("Hello from Deno & grammY!"));

bot.on("message", (ctx) => ctx.reply("How can I help you?"));

bot.start();
```

> > Note: Get your bot token with [BotFather on telegram](https://t.me/BotFather), and set is as an environment variable `TELEGRAM_BOT_TELEGRAM` in Zeabur. 

Reviewer: ```suggestion
> Note: Get your bot token with [@BotFather](https://t.me/BotFather) on Telegram, and set is as an environment variable `TELEGRAM_BOT_TELEGRAM` in Zeabur. 
```
```suggestion
> Note: Get your bot token with [BotFather on telegram](https://t.me/BotFather), and set is as an environment variable `TELEGRAM_BOT_TELEGRAM` in Zeabur. 
```

>
> You can check out [this tutorial](https://zeabur.com/docs/deploy/variables) for setting environment variables in Zeabur.

Then run the following command to start your bot:

```sh
deno run --allow-net main.ts
```

> Deno will automatically download the dependencies, generate the lock file and start your bot.

Reviewer: Style nitpick (both works but I like this better)
```suggestion
Deno will automatically download the dependencies, generate the lock file, and start your bot.
```
